### PR TITLE
feat(snapshot): add snapshot provisioner code

### PR DESCRIPTION
**Pull request by shubham14bajpai**
Signed-off-by: shubham <shubham.bajpai@mayadata.io>

This PR
 - adds the snapshot provisioner code 
 - updates the k8s dependencies to v1.20

Successfully tested the clone bdd on k8s 1.20 cluster

### CODE REVIEW

File: Makefile.buildx.mk [hunk 1]
```diff
@@ -25,7 +25,7 @@ endif
 
 # default list of platforms for which multiarch image is built
 ifeq (${PLATFORMS}, )
-	export PLATFORMS="linux/amd64,linux/arm64"
+	export PLATFORMS="linux/amd64"
```
Comment (kmova) [inline Makefile.buildx.mk]: Will arm be not built?

Comment (kmova) [reply Makefile.buildx.mk]: Or is this about default and value will be overwritten from the workflow.. "Build and Push"

Comment (shubham14bajpai) [reply Makefile.buildx.mk]: This Makefile is no longer used for the workflows. We use the `docker/build-push-action` action which takes all such parameters as fields. This Makefile is maintained only for building & pushing the images locally. Missed to correct the platforms after I tested locally.

**Review by prateekpandey14 — COMMENTED** [review]
@shubham14bajpai let's use atleast 1.20 k8s version in travis yaml, currently its use 1.13 version .

**Pull request merged**